sample_consensus/include/pcl/sample_consensus/impl/mlesac.hpp

```diff
@@ -224,15 +224,22 @@ pcl::MaximumLikelihoodSampleConsensus<PointT>::computeMedianAbsoluteDeviation (
     distances[i] = ptdiff.dot (ptdiff);
   }
 
-  std::sort (distances.begin (), distances.end ());
-
   double result;
   std::size_t mid = indices->size () / 2;
   // Do we have a "middle" point or should we "estimate" one ?
   if (indices->size () % 2 == 0)
-    result = (sqrt (distances[mid-1]) + sqrt (distances[mid])) / 2;
+  {
+    // Looking at two values instead of one probably doesn't matter because they are mostly barely different, but let's do it for accuracy's sake
```

Comment:
I've seen this approach being applied once before (by none other than you 😄, in sample consensus lmeds). It might make sense to extract this into a generic function in `common` or `sample_consensus`

Reply:
:grin: Yes, I could create a generic function for that, IMO rather in common than in sample_consensus, but in which header? My ideas would be either `common.h`, `centroid.h`, `utils.h`, or `vector_average.h`

Reply:
utils?

Reply:
I created it in `common.h` because there are functions that do mean calculations, and `utils.h` seems to be for other things currently.
However, the median computation in lmeds and one of the median computations in mlesac can't use this new function because they take the sqrt of the values, and `( sqrt(a) + sqrt(b) )/2` is not `sqrt( (a+b)/2 )`

Reply:
That can be resolved by taking a `Functor`:
```cpp
/**
 * If the number of samples is odd, finds the median values, passes it through the functor and returns the result.
 * Otherwise averages the two values around the center after applying an optional preprocessing
 */
template<typename real, typename Functor> inline auto pcl::computeMedian (std::vector<real>& values, Functor&& f = [](const real& x){ return x;}) noexcept -> decltype(f(values[0]) {
...
    const auto n_plus_1 = *(std::min_element (values.begin () + mid, values.end()));
    return (f(values[mid-1]) + f(n_plus_1))/2;
...
    return f(values[mid-1]);
}
```
My return type could use some work 😅